linera-execution/src/execution_state_actor.rs

```diff
@@ -360,6 +362,400 @@ where
+                        callback.respond(Err(ExecutionError::UnauthorizedApplication(application_id)));
+                    } else {
+                        write_actor
+                            .system
+                            .application_permissions
+                            .set(application_permissions);
+                        drop(write_actor);
+
+                        callback.respond(Ok(()));
+                    }
+                }
+
+                CreateApplication {
+                    next_message_id,
+                    bytecode_id,
+                    parameters,
+                    required_application_ids,
+                    callback,
+                } => {
+                    let mut write_actor = this.write().await;
+                    let create_application_result = write_actor
+                        .system
+                        .create_application(
+                            next_message_id,
+                            bytecode_id,
+                            parameters,
+                            required_application_ids,
+                        )
+                        .await?;
+                    drop(write_actor);
+
+                    callback.respond(Ok(create_application_result));
+                }
+
+                FetchUrl { url, callback } => {
+                    let bytes = reqwest::get(url).await?.bytes().await?.to_vec();
+                    callback.respond(bytes);
+                }
+
+                HttpPost {
+                    url,
+                    content_type,
+                    payload,
+                    callback,
+                } => {
+                    let res = Client::new()
+                        .post(url)
+                        .body(payload)
+                        .header(CONTENT_TYPE, content_type)
+                        .send()
+                        .await?;
+                    let body = res.bytes().await?;
+                    let bytes = body.as_ref().to_vec();
+
+                    callback.respond(bytes);
+                }
+
+                ReadBlobContent { blob_id, callback } => {
+                    let mut write_actor = this.write().await;
+                    let blob = write_actor.system.read_blob_content(blob_id).await?;
+                    let is_new = write_actor.system.blob_used(None, blob_id).await?;
+
+                    callback.respond((blob, is_new))
+                }
+
+                AssertBlobExists { blob_id, callback } => {
+                    let mut write_actor = this.write().await;
+                    write_actor.system.assert_blob_exists(blob_id).await?;
+                    callback.respond(write_actor.system.blob_used(None, blob_id).await?)
+                }
+            }
+
+            Ok(())
+        };
+
+        // alternative is to use FuturesUnordered
+        let self_transmuted: &'static mut ExecutionStateView<C> = unsafe { std::mem::transmute(self) };
```

Comment:
We try to avoid `unsafe` as much as possible. Maybe `FuturesUnordered` (without spawning Tokio tasks) would work. I need to think a bit more about this :thinking: